mac/deploy_mac.sh

```diff
@@ -41,10 +38,14 @@ cleanup()
 
 build_app()
 {
+    local client_or_server="${1}"
+
     # Build Jamulus
-    qmake "${project_path}" -o "${build_path}/Makefile" "CONFIG+=release" ${@:2}
-    local target_name=$(sed -nE 's/^QMAKE_TARGET *= *(.*)$/\1/p' "${build_path}/Makefile")
-    local job_count="$(sysctl -n hw.ncpu)"
+    qmake "${project_path}" -o "${build_path}/Makefile" "CONFIG+=release" "${@:2}"
+    local target_name
+    target_name=$(sed -nE 's/^QMAKE_TARGET *= *(.*)$/\1/p' "${build_path}/Makefile")
```

Comment:
Needs quotes around the whole `$(...)` expression.

Reply:
I don't think the quotes are needed as they are [implicit for variable assignments](https://stackoverflow.com/questions/46376773/bash-is-subshell-output-implicitly-quoted/46376928#46376928). I'll add them anyway.

Doing all of this doesn't increase my liking of bash/shell scripting, btw...

Reply:
The stupid one (i.e. bash is stupid) is the not allowing `declare -r var=$(get a result)` because the success/failure of `get` is lost.  Bash should prefer the outcome of the subshell over the outcome of the assignment...  I find the two line method annoying ... but necessary now it's been pointed out.

Reply:
I fully agree, but I guess we have to take the `bash` which exists. :)

So, how should we continue here?
- Would you prefer quotes around the subshell invocations in variable assignments?
- Do we need to change anything regarding multi-line assignments (I don't see any alternatives)?

Reply:
I'd leave the quotes off, as they're unnecessary - let bash do the work.  Same with the multi-line assigns: if bash isn't doing the work, we have to live with that, too.

In the PR title, is "spellcheck" misspelled (i.e. `shellcheck`)?

Reply:
> I'd leave the quotes off, as they're unnecessary - let bash do the work. Same with the multi-line assigns: if bash isn't doing the work, we have to live with that, too.

Ok, done.

> In the PR title, is "spellcheck" misspelled (i.e. shellcheck)?

The tool is called `shellcheck`, not `spellcheck`-- is that what you mean? Maybe I'm not seeing any other typo...

Reply:
It's been corrected since I wrote this.